cmd/gaia/app/app.go

```diff
@@ -91,11 +91,11 @@ func NewGaiaApp(logger log.Logger, db dbm.DB, traceStore io.Writer, baseAppOptio
 	// add handlers
 	app.coinKeeper = bank.NewKeeper(app.accountMapper)
 	app.ibcMapper = ibc.NewMapper(app.cdc, app.keyIBC, app.RegisterCodespace(ibc.DefaultCodespace))
-	app.paramsKeeper = params.NewKeeper(app.cdc, app.keyParams)
-	app.stakeKeeper = stake.NewKeeper(app.cdc, app.keyStake, app.coinKeeper, app.RegisterCodespace(stake.DefaultCodespace))
-	app.govKeeper = gov.NewKeeper(app.cdc, app.keyGov, app.paramsKeeper.Setter(), app.coinKeeper, app.stakeKeeper, app.RegisterCodespace(gov.DefaultCodespace))
+	app.paramsKeeper = params.NewKeeper(app.cdc, app.keyParams, app.tkeyParams, nil)
+	app.stakeKeeper = stake.NewKeeper(app.cdc, app.keyStake, app.coinKeeper, app.paramsKeeper.SubStore(stake.DefaultParamSpace), app.RegisterCodespace(stake.DefaultCodespace))
+	app.govKeeper = gov.NewKeeper(app.cdc, app.keyGov, app.paramsKeeper, app.paramsKeeper.SubStore(gov.DefaultParamSpace), app.coinKeeper, app.stakeKeeper, app.RegisterCodespace(gov.DefaultCodespace))
```

Comment:
If Gov takes app.paramsKeeper, it shouldn't need to take in app.paramsKeeper.SubStore, or at least it's not providing any more security since SubStore only takes a string.  With the above modification (ParamSpaceKey instead of string), maybe we can pass in to GovKeeper the following:  paramsKeeper, govParamSpaceKey, []ParamSpaceKey{...}  where the last slice is all the param spaces that this governance module can modify in the future via governance proposals.

See my other comments below on s/params.Keeper.SubStore/params.Keeper.Subspace/g

Reply:
It is possible that the app writer would want to allocate different subspace name to the main gov module, but that is a minor case so I agree we can remove subspace argument there.

About the keys those the gov module is allowed to access, for the sake of extensibility, I think it is better to let the gov receive the master permission over the whole paramstore, because it will be hard to modify then. Instead, gov module can store the subspaces list that it is allowed to access without hard-coded permission, and we can modify that list with parameter change proposal(or anything that is simpler than software upgrade).